### PR TITLE
JWT AUTHENTICATION

### PR DESCRIPTION
**Pull request by MChristian00**
#### What does this PR do?
Add authentication.
#### Description of Task to be completed?
Generate token upon user login and signup.
#### How should this be manually tested?
Clone the repo and change directory into it
Run npm install
Run npm run dev
Open postman, and test the route for login.
#### Any background context you want to provide?
Using MongoDB and Mongoose(ORM) for database functionality.
Using the JWT for authorization not yet implemented.
#### Screenshots (if appropriate)
Login with JWT
Signup with JWT

### CODE REVIEW

**Review by Muhire-Josue — COMMENTED** [review]
Are you using a database? if yes, please mention it on your PR description.

File: src/Helpers/generateToken.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import jwt from "jsonwebtoken";
+
+// console.log(process.env.JWT_KEY);
```
Comment (Muhire-Josue) [inline src/Helpers/generateToken.js]: Please remove uninformative comments

File: src/Helpers/generateToken.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+export const generateToken = (req, res, next) => {
+  let token = jwt.sign(req.body, "SECRET_KEY");
```
Comment (Muhire-Josue) [inline src/Helpers/generateToken.js]: Could you store your secret key in `.env` file?

File: src/Middlewares/Validation/index.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        return res.status(400).json({
+          Error: error.details[0].message,
+        });
+      else next();
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  static async blogUpdateFormValidation(req, res, next) {
```
Comment (Muhire-Josue) [inline src/Middlewares/Validation/index.js]: Please put the `user validation functions` in a separate file, and `blog validation functions` in another separate file.

File: src/app.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+// Routes
+app.use("/api/queries", QueryRoutes);
+app.use("/api/blogs", BlogRoutes);
+app.use("/api/auth", UserRoutes);
+app.use("/api/subscribe", SubscriptionRoutes);
+
+const PORT = process.env.PORT || 3000;
+app.listen(PORT, () => {
+  console.log(`Server started on PORT ${PORT}`);
+});
```
Comment (Muhire-Josue) [inline src/app.js:45]: `export default app;` add this at the end of this file.

**Review by Muhire-Josue — COMMENTED** [review]
Great work christian. please go through the feedback I left you.

File: src/Helpers/generateToken.js (hunk 2 above)
Comment (MChristian00) [reply src/Helpers/generateToken.js:6]: I'm having trouble with storing and accessing the env variables that's why I used a string while I'm still trying to figure out the problem.

File: src/Routes/UserRoutes.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        },
+        (err, user) => {
+          if (err) return res.status(500).send(err);
+          res.status(201).json({
+            Message: "User added",
+            User: user,
+          });
+        }
+      );
+    } catch (error) {
```
Comment (kipropbrian) [inline src/Routes/UserRoutes.js]: Please move functionality to controller files.

File: src/Database/Models/Blog.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        trim: true,
+        required: true,
+      },
+      Picture: {
+        type: URL,
+      },
+      Comments: { type: Array },
+      //   viewsCount: { type: Number, default: 0 },
+      //   LikedBy: { type: Array },
+    },
```
Comment (kipropbrian) [inline src/Database/Models/Blog.js]: Try and avoid commenting on code. Delete it if you do not need it

**Review by MChristian00 — COMMENTED** [review]
All requests from reviewers have been successfully implemented

**Review by Muhire-Josue — APPROVED** [review]
Great work, please go ahead and merge the PR.

**Pull request merged**